flocker/control/_persistence.py

```diff
@@ -163,15 +245,51 @@ def __init__(self, reactor, path):
     def startService(self):
         if not self._path.exists():
             self._path.makedirs()
-        self._config_path = self._path.child(b"current_configuration.v1.json")
+        self.load_configuration()
+        MultiService.startService(self)
+        _LOG_STARTUP(configuration=self.get()).write(self.logger)
+
+    def load_configuration(self):
+        """
+        Load the persisted configuration, upgrading the configuration format
+        if an older version is detected.
+        """
+        self._config_path = self._path.child(b"current_configuration.json")
+        v1_config_path = self._path.child(
+            b"current_configuration.v1.json")
+        # Check for a v1 config and upgrade to latest if found.
+        if v1_config_path.exists():
+            if not self._config_path.exists():
```

Comment:
It would probably be reasonable to either:

  * Move the v1 configuration out of the way as part of the upgrade process, ensuring it doesn't get picked up in the future
  * Reverse the logic here so that newer configuration files are always preferred over older ones

A third option, delete the old configuration, might also be reasonable but I'm happy to err on the side of keeping stuff around for longer than necessary if it helps us avoid even one accidental loss if information.